### PR TITLE
add tkctl check in e2e test

### PR DESCRIPTION
**Pull request by yylt**
### What problem does this PR solve? 
tkctl can not run in pod , so move tkctl into e2e image 

### Check List 

Tests 

Unit test
E2E test
Stability test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has Helm charts change

Has Go code change
Has CI related scripts change

 - Has documents change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:
 
 ```release-note

 ```

### CODE REVIEW

**Comment by Yisaer** [comment]
ref: https://github.com/pingcap/tidb-operator/issues/1067

**Comment by Yisaer** [comment]
@yylt  Hi, do you still continue to work on this issue currently?

**Comment by yylt** [comment]
@Yisaer sorry, I think the question had been solved by this PR , but it must confilcted now, if no one work on this , i can submit new pr
 
I am confused about "build failed message" , could you tell me how to solve this by "failed message"

**Comment by Yisaer** [comment]
@yylt We used `Github in Actions` in the past, and currently removed it in our tools. So you don't have to care about this failure message.

**Comment by cofyc** [comment]
now we can write a ginkgo e2e case in the new framework, here are some examples: https://github.com/pingcap/tidb-operator/blob/master/tests/e2e/tidbcluster/tidbcluster.go

Kubernetes also has many more good e2e tests of `kubectl` we can refer to: https://github.com/kubernetes/kubernetes/blob/master/test/e2e/kubectl/kubectl.go

File: tests/actions_tkctl.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
+package tests
```
[the remark this answers is not in the document]
Comment (yylt) [reply tests/actions_tkctl.go]: Thank you!

**Comment by yylt** [comment]
@Yisaer Hi, Could you review this again? thank you

File: tests/e2e/tidbcluster/tidbcluster.go [hunk 2]
```diff
@@ -59,6 +59,7 @@ import (
 	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
 	"k8s.io/kubernetes/test/utils"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"github.com/cockroachdb/cockroach/pkg/acceptance/localcluster/tc"
```
Comment (cofyc) [inline tests/e2e/tidbcluster/tidbcluster.go]: @yylt I guess it's a mistake made by your editor ;-)

File: tests/actions_tkctl.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+	glog "k8s.io/klog"
+)
+
+var (
+	t = MyTestT{}
+)
+
+type MyTestT struct{}
+
+func (mt *MyTestT) Errorf(format string, args ...interface{}) {
```
Comment (cofyc) [inline tests/actions_tkctl.go]: `MyTestT` is not used?

File: tests/e2e/tidbcluster/tidbcluster.go [hunk 4]
```diff
@@ -686,6 +687,16 @@ var _ = ginkgo.Describe("[tidb-operator] TiDBCluster", func() {
...
 	})
+
+	ginkgo.It("tkctl test cmds [use list] ", func() {
+		cluster := newTidbClusterConfig(e2econfig.TestConfig, ns, "restarter", "admin", "")
+		cluster.Resources["pd.replicas"] = "3"
+		cluster.Resources["tikv.replicas"] = "3"
+		cluster.Resources["tidb.replicas"] = "2"
+		oa.DeployTidbClusterOrDie(&cluster)
+		oa.CheckTidbClusterStatusOrDie(&cluster)
+		oa.CheckTkctlOrDie(&cluster)
```
Comment (cofyc) [inline tests/e2e/tidbcluster/tidbcluster.go]: It's not necessary to add a method `CheckTkctlOrDie` on `operatorActions`. Can you move `CheckTkctlOrDie` implementation in this code block?

File: tests/e2e/tidbcluster/tidbcluster.go (hunk 2 above)
Comment (yylt) [reply tests/e2e/tidbcluster/tidbcluster.go]: It is strange, I don't find this in my commits .

**Comment by yylt** [comment]
@Yisaer  so sorry! I'm so busy this week.

**Comment by yylt** [comment]
@Yisaer  so sorry! I'm so busy this week.

**Pull request closed without merging**